grails-app/services/org/pih/warehouse/data/ProductSupplierIdentifierService.groovy

```diff
@@ -0,0 +1,30 @@
+package org.pih.warehouse.data
+
+import org.pih.warehouse.core.IdentifierService
+import org.pih.warehouse.core.identification.IdentifierGeneratorParams
+import org.pih.warehouse.product.ProductSupplier
+
+class ProductSupplierIdentifierService extends IdentifierService {
+
+    @Override
+    String getEntityKey() {
+        return "productSupplier"
+    }
+
+    @Override
+    protected Integer countDuplicates(String code) {
+        return ProductSupplier.countByCode(code)
+    }
+
+    /**
+     * Generates a new product supplier code.
+     */
+    String generate(String productCode, String organizationCode) {
+        return generate(IdentifierGeneratorParams.builder()
+                .customKeys([
+                        'productCode': productCode,
+                        'organizationCode': organizationCode,
```

Comment:
Seeing customKeys being used for the first time, I'm wondering if there's a more intutive name for that. customKeys implies that we need to pass in a list of map keys. 

You can leave it for now, but something as simple as data or customData might suffice.

Reply:
I like customProperties